### PR TITLE
docs: scaffolding with docusaurus

### PR DESCRIPTION
**Pull request by richardcase**
This scaffolds the initial site using docusaurus. And it includes GHA workflows to publish to GitHub Pages.

### CODE REVIEW

File: docusaurus.config.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+const darkCodeTheme = require('prism-react-renderer/themes/dracula');
+
+/** @type {import('@docusaurus/types').Config} */
+const config = {
+  title: 'Rancher Turtles',
+  tagline: 'Rancher Turtles is an extension to Rancher that brings increased integration with Cluster API.',
+  favicon: 'img/favicon.ico',
+
+  // Set the production url of your site here
+  url: 'https://your-docusaurus-test-site.com',
```
Comment (nunix) [inline docusaurus.config.js:14]: to be modified to the docs site URL

File: docusaurus.config.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    [
+      'classic',
+      /** @type {import('@docusaurus/preset-classic').Options} */
+      ({
+        docs: {
+          sidebarPath: require.resolve('./sidebars.js'),
+          // Please change this to your repo.
+          // Remove this to remove the "edit this page" links.
+          editUrl:
+            'https://github.com/facebook/docusaurus/tree/main/packages/create-docusaurus/templates/shared/',
```
Comment (nunix) [inline docusaurus.config.js:45]: default URL, should be set to project repo or removed

File: docusaurus.config.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        },
+        blog: {
+          showReadingTime: true,
+          // Please change this to your repo.
+          // Remove this to remove the "edit this page" links.
+          editUrl:
+            'https://github.com/facebook/docusaurus/tree/main/packages/create-docusaurus/templates/shared/',
```
Comment (nunix) [inline docusaurus.config.js:52]: default URL, should be set to project repo or removed

File: docusaurus.config.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+  themeConfig:
+    /** @type {import('@docusaurus/preset-classic').ThemeConfig} */
+    ({
+      // Replace with your project's social card
+      image: 'img/docusaurus-social-card.jpg',
+      navbar: {
+        title: 'Rancher Turtles',
+        logo: {
+          alt: 'My Site Logo',
```
Comment (nunix) [inline docusaurus.config.js:69]: Can be set to the project name

File: blog/authors.yml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+richardcase:
+  name: Richard Case
+  title: Maintainer of Docusaurus
```
Comment (nunix) [inline blog/authors.yml:3]: Might be changed to your title

**Review by nunix — COMMENTED** [review]
Made some comments on the config file based on the standard deployments we have for the other docs.

Potentially, add already the gtag > trackingID is available/needed in `docusaurus.config.js`.
There's 2 lockfiles from npm (`package-lock.json`) and yarn (yarn.lock), the best practice is to use only one package manager.

**Comment by richardcase** [comment]
Thanks for the review @nunix.

File: docusaurus.config.js (hunk 1 above)
Comment (richardcase) [reply docusaurus.config.js]: Changed.

File: docusaurus.config.js (hunk 2 above)
Comment (richardcase) [reply docusaurus.config.js]: Changed

File: docusaurus.config.js (hunk 4 above)
Comment (richardcase) [reply docusaurus.config.js]: Changed

**Review by furkatgofurov7 — APPROVED** [review]
Great work!

CI needs a fixing, but overall LGTM

File: docusaurus.config.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+const darkCodeTheme = require('prism-react-renderer/themes/dracula');
+
+/** @type {import('@docusaurus/types').Config} */
+const config = {
+  title: 'Rancher Turtles',
+  tagline: 'Rancher Turtles is an extension to Rancher that brings increased integration with Cluster API.',
+  favicon: 'img/favicon.ico',
+
+  // Set the production url of your site here
+  url: 'https://rancher-sandbox.github.io/rancher-turtles-docs/',
```
Comment (Danil-Grigorev) [inline docusaurus.config.js:14]:
```suggestion
  url: 'https://rancher-sandbox.github.io/',
```

File: docusaurus.config.js [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+  // Set the /<baseUrl>/ pathname under which your site is served
+  // For GitHub pages deployment, it is often '/<projectName>/'
+  baseUrl: '/',
```
Comment (Danil-Grigorev) [inline docusaurus.config.js:17]:
```suggestion
  baseUrl: '/rancher-turtles-docs/',
```

**Review by Danil-Grigorev — COMMENTED** [review]
With that build and deploy should pass, LGTM.

File: docusaurus.config.js (hunk 6 above)
Comment (richardcase) [reply docusaurus.config.js]: Changed

File: docusaurus.config.js (hunk 7 above)
Comment (richardcase) [reply docusaurus.config.js]: Changed

**Review by Danil-Grigorev — APPROVED** [review]

**Pull request merged**